apps/omg_watcher_info/lib/omg_watcher_info/block_application_consumer.ex

```diff
@@ -14,45 +14,51 @@
 
 defmodule OMG.WatcherInfo.BlockApplicationConsumer do
   @moduledoc """
-  Subscribes for new blocks and inserts them to WatcherInfo.DB.
+  Subscribes for new blocks and inserts them into a pending queue of blocks waiting to be processed.
   """
-  alias OMG.WatcherInfo.DB
   require Logger
 
-  ### Client
-
-  def start_link(_args) do
-    GenServer.start_link(__MODULE__, :ok, name: __MODULE__)
-  end
+  use GenServer
 
-  ### Server
+  alias OMG.WatcherInfo.DB
 
-  use GenServer
+  @default_bus_module OMG.Bus
 
-  def init(:ok) do
-    :ok = OMG.Bus.subscribe({:child_chain, "block.get"}, link: true)
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
 
+  def init(args) do
     _ = Logger.info("Started #{inspect(__MODULE__)}")
+
+    bus_module = Keyword.get(args, :bus_module, @default_bus_module)
+    :ok = bus_module.subscribe({:child_chain, "block.get"}, link: true)
+
     {:ok, %{}}
   end
 
   # Listens for blocks and insert them to the WatcherDB.
   def handle_info({:internal_event_bus, :block_received, block_application}, state) do
-    _ =
+    {:ok, _} =
       block_application
-      |> to_mined_block()
-      |> DB.Block.insert_with_transactions()
+      |> to_pending_block()
+      |> DB.PendingBlock.insert()
```

Comment:
`PendingBlock` is somewhat different naming then I would expect. the world `queue` is not used anywhere :)))